### ISSUE TITLE
Refactor `TrackingDataset` structure and RUN_DATA schema

### ISSUE THREAD

**Issue by Huongg** [kedro-org/kedro-viz#1004]
## Description

The structure of the `TrackingDataset` is increasingly difficult to work with. Further, it forces us to create several loops in the frontend to property setup and layout the page. 

A better shape and structure should be found.

## Checklist

- [x] Include labels so that we can categorise your feature request

**Comment by AntonyMilneQB** [comment]
tl;dr: excellent idea. Very happy to do this, but I would like to get a more concrete vision of what the ideal state of the GraphQL schema should eventually look like before making any changes. This would ensure that implementing this will take us closer to that ideal state rather than further from it.

---

I think this is a very good idea and I'm sure would be feasible on the backend. I suspect that one of the reasons these are in separate queries is because the data comes from two different places: metadata from the SQLite database (`data_access_manager.runs`) and tracking datasets from the file system (`data_access_manager.tracking_datasets`). But as far I can see that's not a good reason to keep them as separate queries. It should be the job of the backend to get data from different repositories and deliver it to the frontend in the required format.

Going beyond this, I think the whole data structure here needs some rethinking. Here's the relevant part of [the schema](https://github.com/kedro-org/kedro-viz/blob/main/.github/img/schema.graphql.png)
<img width="1180" alt="image" src="https://user-images.githubusercontent.com/49395058/183597041-a358b6cd-a0d5-4e16-96db-7007ab783bf3.png">

I don't like the `TrackingDataSet` structure here:
* the `data` field is a nested structure just given as `JSON`. The structure of this should be provided by GraphQL types, not just a generic JSON object which conveys no semantic meaning. If made less generic, we could implement the different tracking types (plot, json, metrics) much more cleanly, e.g by using GraphQL interfaces
* why are `run_ids` so deeply buried inside the `data` structure? Just the other day @tynandebold needed to make a new field here to extract the `runIds` to a top-level field.

As I understand it, the current structure of `TrackingDataSet` was driven by constraints on the FE that needed to compare the same dataset across multiple runs + toggle the "show diff" button to show/hide _rows_ of the table. But it doesn't feel natural to me from a backend perspective at all.

So if we could rethink the whole structure of this part of the GraphQL schema then that would be great. This is way beyond the scope of combining the two queries you mention, but I think it would be good to first get some idea of what our ideal final state looks like. There's no need to implement it all in one go, and combining the two queries you mention sounds like a great place to start (especially given there's an immediate need to do so in order to solve your problem). Just I want to avoid patching together a schema that will solve one particular problem but accidentally take us further from that ideal state and need further patching in future, e.g. we don't want to end up with `runIds` somehow being a field in 5 different places.

**Comment by tynandebold** [comment]
Agree on all points, Antony. Let's start thinking about what our new schema and structure should be, especially around `TrackingDataSet`. That's always been a thorn in our side.

For combining queries, @rashidakanchwala came up with a great interim solution in #1006 that doesn't change any backend code at all and instead groups our queries in the frontend.

**Comment by tynandebold** [comment]
Note: with the quick solve of merging the two `metadata` and `trackingData` queries thanks to #1006, I've renamed this ticket and updated the description to align more with Antony's great comment above.

**Comment by tynandebold** [comment]
The outcome of this ticket should be the newly agreed upon schema/shape of the data for experiment run metadata and run tracking data.

**Comment by tynandebold** [comment]
Here's an first go at what a new structure could look like, combining our current two queries into one and simplifying in a such a way that (hopefully) makes sense:

```jsonc
{
  "runMetadata": {
    "runIds": ["run123", "run456"],
    "createdBy": ["huong", "rashida"],
    "notes": ["This is a note", "This is a different note"]
  },
  "runTrackingData": {
    "runIds": ["run123", "run456"],
    "metrics": [ // Top-level accordion
      {
        "train_evaluation.r2_score_linear_regression": [ // Dataset accordion
          {
            "copy_X": [123, 312] // Metric row
          },
          {
            "fit_intercept": [null, 0.234]
          }
        ]
      }
    ],
    "plots": [
      {
        "reporting.price_histogram": [
          {
            "price_histogram.json": [{}, null]
          },
          {
            "feature_importance_plot.json": [{}, null]
          }
        ]
      }
    ]
  }
}
```

One of the toughest problems of the current structure is that we're never sure what the row labels will be for the metrics. We do a lot of work to mix and match their placement and even their existence. The thinking here is we move away from that to a row-based approach instead of the current column-based one. We'll no longer need a lot of the FE logic that checks `if (i === 0)` when we need to render a label versus a value. The backend will also now send us `null` values because we know for certain that we need to render them on the frontend.

**Comment by tynandebold** [comment]
Looking at this again, I don't think my proposed structure in the previous comment is good enough. We certainly still need `datasetType` in there somewhere, and the changing of the `runMetadata` probably isn't necessary. Will need to take another stab at this.
